lib/Backends/CPU/FunctionSpecializer.cpp

```diff
@@ -267,7 +269,7 @@ class FunctionSpecializer {
     for (auto &arg : call->arg_operands()) {
       auto curArgIdx = argIdx++;
 
-      if (!shouldSpecializeParameter(arg)) {
+      if (curArgIdx > 63 || !shouldSpecializeParameter(arg)) {
```

Comment:
Looking almost good. I'd just move this new check into `shouldSpecializeParameter`, because it would be cleaner.

Right now `shouldSpecializeParameter` takes only the `arg` as its arguments. But it seems like it should also take parameters called `callee` and `argIdx` with values `callee` and `curArgIdx` respectively. This way more complex conditions can be expressed inside `shouldSpecializeParameter` now and in the future.